builtin/rebase--helper.c

```diff
@@ -73,18 +79,59 @@ int cmd_rebase__helper(int argc, const char **argv, const char *prefix)
 	flags |= abbreviate_commands ? TODO_LIST_ABBREVIATE_CMDS : 0;
 	flags |= rebase_merges ? TODO_LIST_REBASE_MERGES : 0;
 	flags |= rebase_cousins > 0 ? TODO_LIST_REBASE_COUSINS : 0;
-	flags |= command == SHORTEN_OIDS ? TODO_LIST_SHORTEN_IDS : 0;
 
 	if (rebase_cousins >= 0 && !rebase_merges)
 		warning(_("--[no-]rebase-cousins has no effect without "
 			  "--rebase-merges"));
 
+	if (command == MAKE_SCRIPT && !upstream && squash_onto) {
+		FILE *squash_onto_file;
+		int ret;
+
+		squash_onto_file = fopen(path_squash_onto(), "w");
+		if (squash_onto_file == NULL)
+			return error_errno(_("could not open '%s'"), path_squash_onto());
+
+		ret = fputs(squash_onto, squash_onto_file);
+		fclose(squash_onto_file);
+
+		if (ret < 0)
+			return error_errno(_("could not write to '%s'"), path_squash_onto());
+	}
+
+	if (command == COMPLETE_ACTION || command == MAKE_SCRIPT) {
+		const char *shortupstream, *shorthead;
+		struct object_id orig_head;
+
+		if (get_oid("HEAD", &orig_head))
+			die(_("No HEAD?"));
+
+		head_hash = find_unique_abbrev(&orig_head, GIT_MAX_HEXSZ);
+		shorthead = find_unique_abbrev(&orig_head, DEFAULT_ABBREV);
+
+		if (upstream) {
+			struct object_id upstream_oid;
+
+			get_oid(upstream, &upstream_oid);
+			shortupstream = find_unique_abbrev(&upstream_oid, DEFAULT_ABBREV);
+			strbuf_addf(&revisions, "%s...%s", upstream, head_hash);
+			strbuf_addf(&shortrevisions, "%s..%s", shortupstream, shorthead);
+		} else {
+			strbuf_addf(&revisions, "%s...%s", onto, head_hash);
+			strbuf_add(&shortrevisions, shorthead, strlen(shorthead));
+		}
+	}
+
```

Comment:
In my opinion, it would make the code more readable to have this refactored out into its own function, which is called precisely before it is needed, i.e. directly before the calls to `sequencer_*()`.

This would also make it easier to take custody of the memory. And *then* `xstrfmt()` would make a ton of sense. Something like

```c
static int get_revision_ranges(const char *upstream, const char *onto, char **head_hash, char **revisions, char **shortrevisions)
{
...
}
```

Then, you would call that function in the below code, with variables local to that scope of the conditional block, assign the return value of the call into the `sequencer_*()` function to a local variable, release the memory, return the value, clap your hands, done.